### ISSUE TITLE
Failed to infer correct type argument of method invocation

### ISSUE THREAD

**Issue by CharlesZ-Chen** [typetools/checker-framework#1526]
Test case:

```java
public class C {
    public <T> T get(T t) {
            return this.get(t);
    }
}
```

Running Checker Framework (version 2.0) with any checker would raise below type check errors:

```
C.java:3: error: [return.type.incompatible] incompatible types in return.
            return this.get(t);
                           ^
  found   : @UnknownKeyFor Object
  required: T[ extends @UnknownKeyFor Object super @KeyForBottom Void]
C.java:3: error: [return.type.incompatible] incompatible types in return.
            return this.get(t);
                           ^
  found   : @Initialized @Nullable Object
  required: T extends @Initialized @Nullable Object
2 errors
```

It is expected to infer return type of `this.get(t)` to be `T`, but somehow CF infer it as `Object`, result in above unexpected type check errors.
